mqtt/mqtt-bridge/src/persist/loader.rs

```diff
@@ -78,9 +77,34 @@ where
 
             // refresh next batch
             // if error, either someone forged the database or we have a database schema change
-            let next_batch = self.next_batch()?;
+            let mut new_batch = self.next_batch()?;
+
+            // drop those loaded keys which do not exist in the new batch
             let mut inner = self.0.lock();
-            inner.batch = next_batch;
+            if let Some((new_key, _)) = new_batch.front() {
+                while inner.loaded.front().map_or(false, |key| new_key != key) {
+                    inner.loaded.pop_front();
+                }
+            }
```

Comment:
So this works by looking linearly through the batch and removing duplicates.

This is one example you showed me:
```
0
1
1
2
2
3
3
4
4
5
5
6
5
6
7
5
6
7
8
5
```

It would seem there might be cases based on this example where we might get duplicates still?

Reply:
Looking through more carefully, I think we are good.

Reply:
so what happened before:
1. got a batch [0]
2. send a message 0
3. got next batch [1], send 1 but not quite fast as 0
4. got next batch [1, 2], oh, lets send 1, then 2
5. 1 is send, got next batch [2, 3], oh lets send 2, then 3
6. ....

Now because I assume that there is no bugs in RB (can I rely on that?) messages will come ordered and w/o dups. I just remove the prefix of messages already being sent and removed from the storage. next I remove those were being sent, but not removed yet. Add the rest of message as a new batch and continue polling items

Reply:
If you get [1] in step 3, why does next batch again include "1"? That doesn't seem right... what am I missing?